### PR TITLE
Implemented option for Lasso to read primary channel from a file

### PR DESCRIPTION
**Pull request by wutmeownow**
Removed datafind method that caused issues with with CI.

### CODE REVIEW

File: gwdetchar/lasso/tests/test_main.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# global test objects
+
+# volatile temp file to write expected timeseries to
+TEMP = tempfile.NamedTemporaryFile()
+TEST_CHANNEL = 'L1:DMT-SNSW_EFFECTIVE_RANGE_MPC.mean'
+TEST_START = 0
+TEST_END = TEST_START+1000
+TEST_ARRAY = np.linspace(TEST_START, TEST_END, num=TEST_END-TEST_START)
+expected_ts = TimeSeries(TEST_ARRAY, t0=TEST_START, dt=1, channel=TEST_CHANNEL)
+expected_ts.write(TEMP, format='gwf')
```
Comment (duncanmmacleod) [inline gwdetchar/lasso/tests/test_main.py:25]: This should be moved into a fixture function to be evaluated only when the test that requires it is called:

```suggestion
TEST_CHANNEL = 'L1:DMT-SNSW_EFFECTIVE_RANGE_MPC.mean'
TEST_START = 0
TEST_END = TEST_START+1000


@pytest.fixture
def primary_ts_file(tmp_path):
    # construct random data to read
    data = np.linspace(TEST_START, TEST_END, num=TEST_END-TEST_START)
    ts = TimeSeries(data, t0=TEST_START, dt=1, channel=TEST_CHANNEL)
    
    # write data to a file and return that
    outfile = tmp_path / "data.gwf"
    ts.write(outfile, format='gwf')
    return outfile
```

File: gwdetchar/lasso/tests/test_main.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+def test_read():
+    # test reading primary TimeSeries file
+    # try finally for closing temp file
+    try:
+        actual_ts = lasso.get_primary_ts(filepath=TEMP, channel=TEST_CHANNEL,
+                                         start=TEST_START, end=TEST_END,
+                                         cache=None, nproc=1)
+    finally:
+        TEMP.close
```
Comment (duncanmmacleod) [inline gwdetchar/lasso/tests/test_main.py:38]: If the above is transformed into a fixture, then the following changes need to be made here:

```suggestion
def test_read(expected_ts, expected_ts_file):
    actual_ts = lasso.get_primary_ts(
        filepath=expected_ts_file,
        channel=TEST_CHANNEL,
        start=TEST_START,
        end=TEST_END,
        cache=None,
        nproc=1,
    )
```

File: gwdetchar/lasso/tests/test_main.py (hunk 1 above)
Comment (eagoetz) [reply gwdetchar/lasso/tests/test_main.py]: This looks like it has been resolved following Duncan's suggestion

File: gwdetchar/lasso/tests/test_main.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    try:
+        actual_ts = lasso.get_primary_ts(
+            filepath=expected_ts_file,
+            channel=TEST_CHANNEL,
+            start=TEST_START,
+            end=TEST_END,
+            cache=None,
+            nproc=1)
+    except Exception as e:
+        print(e)
```
[the remark this answers is not in the document]
Comment (eagoetz) [reply gwdetchar/lasso/tests/test_main.py]: This has been removed.

File: gwdetchar/lasso/tests/test_main.py (hunk 2 above)
Comment (eagoetz) [reply gwdetchar/lasso/tests/test_main.py]: This has been resolved.

**Comment by eagoetz** [comment]
Thanks for working on this @wutmeownow and thanks @duncanmmacleod for your helpful insights. We are approving and merging this now.

**Review by eagoetz — APPROVED** [review]
LGTU

**Pull request merged**